py/embedded-server/java-runtime/src/main/java/io/deephaven/python/server/EmbeddedServer.java

```diff
@@ -0,0 +1,105 @@
+package io.deephaven.python.server;
+
+import dagger.Component;
+import io.deephaven.configuration.Configuration;
+import io.deephaven.engine.util.ScriptSession;
+import io.deephaven.server.healthcheck.HealthCheckModule;
+import io.deephaven.server.jetty.JettyServerModule;
+import io.deephaven.server.plugin.python.PythonPluginsRegistration;
+import io.deephaven.server.runner.DeephavenApiServer;
+import io.deephaven.server.runner.DeephavenApiServerComponent;
+import io.deephaven.server.runner.DeephavenApiServerModule;
+import io.deephaven.server.runner.Main;
+import io.deephaven.server.util.Scheduler;
+import org.jpy.PyDictWrapper;
+import org.jpy.PyObject;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+import java.io.IOException;
+import java.util.concurrent.TimeoutException;
+
+public class EmbeddedServer {
+    @Singleton
+    @Component(modules = {
+            DeephavenApiServerModule.class,
+            HealthCheckModule.class,
+            PythonPluginsRegistration.Module.class,
+            JettyServerModule.class
+    })
+    public interface PythonServerComponent extends DeephavenApiServerComponent {
+        @Component.Builder
+        interface Builder extends DeephavenApiServerComponent.Builder<PythonServerComponent.Builder> {
+            PythonServerComponent build();
+        }
+
+        void injectFields(EmbeddedServer instance);
+    }
+
+    @Inject
+    DeephavenApiServer server;
+    @Inject
+    Scheduler scheduler;
+    @Inject
+    Provider<ScriptSession> scriptSession;
+
+    public EmbeddedServer(int port, PyObject dict) throws IOException {
+        final Configuration config = Main.init(new String[0], EmbeddedServer.class);
+        PyDictWrapper pyConfig = dict.asDict();
```

Comment:
Nothing yet, just trying to get the api right for future things. Probably can back this part out, until parts of the server can be restarted cleanly.

But since all configuration for the _engine_ must be set no later than the moment the first class loads, we're extremely restricted in what we can configure - and can't detect the scope of any of those settings either.

Any idea of stopping/starting will really depend on entirely stopping the jvm and starting a fresh one, unless we stop using Configuration in :server etc.